### PR TITLE
[4.0] Featured Contact [a11y]

### PR DESCRIPTION
**Pull request by brianteeman**
### Summary of Changes
This PR addresses multiple accessibility, consistency and usability issues with the featured icon in com_contacts.

After reports from webaim and checking with Bruce Lawson it is not safe to rely on a "title" attribute to provide the required text for assistive technology.

In this PR the text from the title is now present as a div instead of a title

The text did not indicate what the current featured status was so it has been changed and is not consistent with the published status icon

### Before
before

### After
after

### CODE REVIEW

**Comment by hans2103** [comment]
@brianteeman 
According to https://www.w3.org/TR/wai-aria-practices/#tooltip

> ## WAI-ARIA Roles, States, and Properties
> - The element that serves as the tooltip container has role `tooltip`.
> - The element that triggers the tooltip references the tooltip element with `aria-describedby`.

Your pull request uses `aria-labelledby` instead of `aria-describedby`. 
Can you change this into `aria-describedby`?

**Comment by brianteeman** [comment]
I'm not sure I agree. The "error" could be the role. If this needs to be changed then it needs to be changed everywhere this pattern is used and I am sure the accessibility team had already checked it. Especially as the first use was over a year ago

**Comment by hans2103** [comment]
> If this needs to be changed then it needs to be changed everywhere this pattern is used...

No problem.... just a lot of PR's with aria-labelledby changing into aria-describedby

> and I am sure the accessibility team had already checked it.

I am just copying he W3.org rules. 
@chmst please join us in this discussion.

**Comment by brianteeman** [comment]
Whatever is correct I am happy to change. Just want the accessibility team to make a decision.

Personally I would merge this one and then they are all consistent and if necessary change them all in another pr

**Comment by brianteeman** [comment]
For reference there are multiple different uses of this pattern. Here it is used because there is no label on the link and I was following the advice of experts so here I think the role is the problem. 

In some other cases such as in the list of available languages it is a description for an icon but has no aria

And finally it is used on the search field but there it is defiitely correctly using described by

**Comment by hans2103** [comment]
@brianteeman someone started an issue on https://github.com/w3c/aria-practices/issues/1034
I've seen other issues in w3c/aria-practices about `aria-labelledby`, `aria-describedby` and even about `role="tooltip"`

In other words.. using `aria-labelledby` or `aria-describedby` should be another discussion. Your PR copies the way J4 implemented tooltip. And that is fine. 

I am going to test your PR

**Comment by brianteeman** [comment]
@hans2103 its kind of off-topic but the aria-practices site is regularly criticised by leading accessibility experts including members of wcag https://adrianroselli.com/2017/10/dont-use-aria-menu-roles-for-site-nav.html

File: administrator/components/com_contact/src/Service/HTML/AdministratorService.php [hunk 1]
```diff
@@ -123,21 +123,22 @@ public function association($contactid)
...
 		);
 		$state = ArrayHelper::getValue($states, (int) $value, $states[1]);
 		$icon = $state[0] === 'featured' ? 'star featured' : 'star';
 
 		if ($canChange)
 		{
 			$html = '<a href="#" onclick="return Joomla.listItemTask(\'cb' . $i . '\',\'' . $state[1] . '\')" class="tbody-icon'
-				. ($value == 1 ? ' active' : '') . '" title="' . Text::_($state[3])
-				. '"><span class="fas fa-' . $icon . '" aria-hidden="true"></span></a>';
+				. ($value == 1 ? ' active' : '') . '" aria-labelledby="cb' . $i . '-desc">'
```
Comment (hans2103) [inline administrator/components/com_contact/src/Service/HTML/AdministratorService.php:139]: When I look at the HTML output of com_contact in administrator I see the attribute `aria-labelledby` on both _State_ as _Featured_. When I take a look at the code I don't see `aria-labelledby` on _State_

https://github.com/joomla/joomla-cms/blob/0606c31fd9c956ac7af241214706b6dd9e82f73e/administrator/components/com_contact/src/Service/HTML/AdministratorService.php#L98-L99

After testing your code and removed the attribute `aria-labelledby` and refresh the administrator/com_contact I notice that the attribute is set. Might this be done by some Javascript?

In other words... I think you can remove `aria-labelledby`. It will be set when the page is rendered.

**Review by hans2103 — COMMENTED** [review]
> For reference there are multiple different uses of this pattern. Here it is used because there is no label on the link and I was following the advice of experts so here I think the role is the problem.

Just found the same argument about using `aria-labelledby` on https://github.com/w3c/aria-practices/issues/1034

File: administrator/components/com_contact/src/Service/HTML/AdministratorService.php (hunk 1 above)
Comment (brianteeman) [reply administrator/components/com_contact/src/Service/HTML/AdministratorService.php:139]: The quoted code is not used for state it is used for associations. State gets it stuff from`libraries\src\HTML\Helpers\JGrid.php`

Comment (brianteeman) [reply administrator/components/com_contact/src/Service/HTML/AdministratorService.php:139]: I can not replicate your findings about removing aria-labelledby. If I remove it from `administrator/components/com_contact/src/Service/HTML/AdministratorService.php` then it is not present for featured. It is still present for state because thats different code as mentioned above

Comment (hans2103) [reply administrator/components/com_contact/src/Service/HTML/AdministratorService.php:139]: damn... caching I'm afraid. 
I can sure tell you I've tested your PR. :-)

**Comment by hans2103** [comment]
I have tested this item :white_check_mark: successfully on d0e0a68b2961706a2567980c92fa746a09363634<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29928">issues.joomla.org/tracker/joomla-cms/29928</a>.</sub>

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on d0e0a68b2961706a2567980c92fa746a09363634<br /><br />My favorite source is the webaim where aria-describedby is used in sample code, but as @hans2103 wrote, it is not necessary what we must decide in this PR. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29928">issues.joomla.org/tracker/joomla-cms/29928</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29928">issues.joomla.org/tracker/joomla-cms/29928</a>.</sub>

**Comment by richard67** [comment]
@brianteeman Please fix PHPCS errors reported by Drone here [https://ci.joomla.org/joomla/joomla-cms/33631/1/9](https://ci.joomla.org/joomla/joomla-cms/33631/1/9):

```
FILE: ...components/com_contact/src/Service/HTML/AdministratorService.php
----------------------------------------------------------------------
FOUND 4 ERRORS AFFECTING 4 LINES
----------------------------------------------------------------------
 126 | ERROR | [x] Line indented incorrectly; expected 1 tabs, found
     |       |     1
     |       |     (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
 127 | ERROR | [x] Opening brace indented incorrectly; expected 5
     |       |     spaces, found 4
     |       |     (PEAR.Functions.FunctionDeclaration.BraceIndent)
 141 | ERROR | [x] Concat operator must be surrounded by a single
     |       |     space
     |       |     (Squiz.Strings.ConcatenationSpacing.PaddingFound)
 150 | ERROR | [x] Closing brace indented incorrectly; expected 5
     |       |     spaces, found 4. Note: the autofixer will convert
     |       |     spaces to tabs
     |       |     (Squiz.WhiteSpace.ScopeClosingBrace.Indent)
```

**Comment by wilsonge** [comment]
Thanks!
